### ISSUE TITLE
Castling indicators in FEN get reset

### ISSUE THREAD

**Issue by Davidy22** [niklasf/python-chess#588]
Using python 3.8, python-chess 0.31.2:

Castling indicators in FEN are reset/corrupted when making moves from certain boards. The following code, adapted from a variant starting board scenario:

```
import chess

b = chess.Board()

highmove = "Qxc5"
b.set_board_fen("k6K/8/8/pppppppp/8/8/8/QqQq4")
print(b.fen())
b.push_san(highmove)
print(b.fen())
```

Gives the output:

```
k6K/8/8/pppppppp/8/8/8/QqQq4 w - - 0 1
k6K/8/8/ppQppppp/8/8/8/Qq1q4 b KQkk - 0 1
```

This is the first specific board and move that I've managed to capture this issue occurring, although I believe I've encountered this same issue prior to this.

To note, I was unable to reproduce this specific issue in the python interactive console.

**Comment by niklasf** [comment]
Thanks for reporting.

**Comment by niklasf** [comment]
Fix released as part of v0.31.3.
